src/client/telemetry/index.ts

```diff
@@ -483,7 +491,38 @@ export interface IEventNamePropertyMapping {
      * Sent when a jupyter session fails to start and we ask the user for a new kernel
      */
     [Telemetry.AskUserForNewJupyterKernel]: never | undefined;
-    [Telemetry.KernelListingPerf]: never | undefined;
+    /**
+     * Time taken to list the Python interpreters.
+     */
+    [Telemetry.InterpreterListingPerf]: {
+        /**
+         * Whether this is the first time in the session.
+         * (fetching kernels first time in the session is slower, later its cached).
+         * This is a generic property supported for all telemetry (sent by decorators).
+         */
+        firstTime?: boolean;
+    };
+    [Telemetry.ActiveInterpreterListingPerf]: {
+        /**
+         * Whether this is the first time in the session.
+         * (fetching kernels first time in the session is slower, later its cached).
+         * This is a generic property supported for all telemetry (sent by decorators).
+         */
+        firstTime?: boolean;
+    };
+    [Telemetry.KernelListingPerf]: {
+        /**
+         * Whether this is the first time in the session.
+         * (fetching kernels first time in the session is slower, later its cached).
+         * This is a generic property supported for all telemetry (sent by decorators).
+         */
+        firstTime?: boolean;
+        /**
+         * Whether this telemetry is for listing of all kernels or just python or just non-python.
+         * (fetching kernels first time in the session is slower, later its cached).
+         */
+        kind: 'remote' | 'local' | 'localKernelSpec' | 'localPython';
```

Comment:
Differentiate between the different telemetry items (whether its all local, remote, or just kernel spec or python)